src/makeCalculator.js

```diff
@@ -37,7 +37,39 @@
  * @return {object}
  */
 function makeCalculator() {
-  // write code here
+  const calculator = {
+    result: 0,
+
+    divide(a, b) {
+      return a / b;
+    },
+
+    add(a, b) {
+      return a + b;
+    },
+
+    subtract(a, b) {
+      return a - b;
+    },
+
+    multiply(a, b) {
+      return a * b;
+    },
+
+    reset() {
+      this.result = 0;
+
+      return this;
+    },
+
+    operate(action, number) {
+      this.result = action(this.result, number);
```

Comment:
let's improve your code, I think we can avoid putting `this.result` directly in our callback.

we can use `bind`, `call` or `apply`  to have proper `this` inside our callback.



[call-apply-and-bind](https://www.freecodecamp.org/news/understand-call-apply-and-bind-in-javascript-with-examples/)